libsql_client/config.py

```diff
@@ -24,11 +27,38 @@ def _expand_config(url: str, *, auth_token: Optional[str]) -> _Config:
     for key, value in qsl:
         if key == "authToken":
             auth_token = value or None
+        elif key == "tls":
+            if value == "0":
+                tls = False
+            elif value == "1":
+                tls = True
+            else:
+                raise LibsqlError(
+                    f"Unknown value for the 'tls' query argument: {value!r}. "
+                    "Supported values are '0' and '1'",
+                    "URL_INVALID",
+                )
         else:
             raise LibsqlError(
                 f"Unknown URL query parameter {key!r}", "URL_PARAM_NOT_SUPPORTED"
             )
 
+    if scheme == "libsql":
+        if tls is False:
+            if url_parsed.port is None:
+                raise LibsqlError(
+                    "A 'libsql:' URL with ?tls=0 must specify an explicit port",
+                    "URL_INVALID",
+                )
```

Comment:
this part hurts usability, but AFAIU we'll be changing turso cmdline to always print the full URL, right?  then if it's a matter of "copy & paste", it shouldn't matter.

that said, I don't see why can't we default to 443/80 based on tls